src/library/scala/collection/SeqLike.scala

```diff
@@ -507,9 +507,8 @@ trait SeqLike[+A, +Repr] extends Any with IterableLike[A, Repr] with GenSeqLike[
     val b = newBuilder
     val seen = mutable.HashSet[A]()
     for (x <- this) {
-      if (!seen(x)) {
+      if (seen.add(x)) {
         b += x
-        seen += x
```

Comment:
In the case that `Set#apply` would have returned `true`, the new formulation actually does a bit more work by updating the set with the new element. So I'd expect the new version to be slightly slower when there are many copies of given elements.

It does appear to do less work in the case that `xs.distinct.size / xs.size` is closer to 1.

Would you care to test out these variations in a benchmark? See #5537 for an good example of a benchmark-driven performance patch.